saylua/models/user.py

```diff
@@ -35,139 +49,151 @@ class _User(db.Model):
-    pronouns = ndb.StringProperty(default='', indexed=False)
-    bio = ndb.StringProperty(default='', indexed=False)
 
-    def get_role(self):
-        return Role.get_by_id(self.role_id)
+class BankAccount(db.Model):
+    """Permanently assigned bank account for the `User` class.
 
-    @classmethod
-    def by_username(cls, username):
-        return cls.query(cls.usernames == username.lower()).get()
+    One to one relationship with `User`.
+    """
 
-    @classmethod
-    def key_by_username(cls, username):
-        return cls.query(cls.usernames == username.lower()).get(keys_only=True)
+    __tablename__ = "bank_accounts"
 
-    @classmethod
-    def hash_password(cls, password, salt=None):
-        if not salt:
-            salt = bcrypt.gensalt()
-        return bcrypt.hashpw(password, salt)
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
+    user = db.relationship("User", back_populates="bank_account")
+    bank_ss = db.Column(db.Integer, default=0)
+    bank_cc = db.Column(db.Integer, default=0)
 
-    @classmethod
-    def check_password(cls, user, password):
-        return cls.hash_password(password, user.phash) == user.phash
 
-    @classmethod
-    @ndb.transactional(xg=True)
-    def transfer_currency(cls, from_key, to_key, cc=0, ss=0):
-        from_user, to_user = ndb.get_multi([from_key, to_key])
-        from_user.star_shards -= ss
-        from_user.cloud_coins -= cc
+class DisplayName(db.Model):
+    """Represents the actual displayed name of a user.
 
-        to_user.star_shards += ss
-        to_user.cloud_coins += cc
+    Multiple names can be tied to one user, but only one of
+    them can be active and nested within the user class.
 
-        # Throw exceptions if the currency amount is invalid
-        cls.except_if_currency_invalid(from_user)
-        cls.except_if_currency_invalid(to_user)
+    One to one relationship with `User`.
+    """
 
-        ndb.put_multi([from_user, to_user])
+    __tablename__ = "display_names"
 
-    @classmethod
-    @ndb.transactional
-    def update_currency(cls, user_key, cc=0, ss=0):
-        user = user_key.get()
-        user.star_shards += ss
-        user.cloud_coins += cc
+    id = db.Column(db.Integer, primary_key=True)
+    active = db.Column(db.Boolean, default=True)
 
-        # Throw exceptions if the currency amount is invalid
-        cls.except_if_currency_invalid(user)
-        user.put()
-        return [user.cloud_coins, user.star_shards]
+    # The linked user
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+    user = db.relationship("User", back_populates="display_name")
 
-    @classmethod
-    def except_if_currency_invalid(cls, user):
-        if user.star_shards < 0 or user.cloud_coins < 0:
-            raise cls.InvalidCurrencyException('Currency cannot be negative!')
+    # The actual display name
+    display_name = db.Column(db.String(80), unique=True)
```

Comment:
Probably makes more sense for this to be `DisplayName.name` rather than `DisplayName.display_name`.